helpers/localize.js

```diff
@@ -0,0 +1,30 @@
+import { html } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+
+export const allowedTags = Object.freeze(['d2l-link', 'd2l-tooltip-help', 'p', 'br', 'b', 'strong', 'i', 'em']);
+
+const markupError = `localizeHTML() rich-text replacements must use markup templates with only the following allowed elements: ${allowedTags}. [link to docs]`;
+const validTerminators = '([>\\s/]|$)';
+const allowedAfterTriangleBracket = `/?(${allowedTags.join('|')})?${validTerminators}`;
+const disallowedTagsRegex = new RegExp(`<(?!${allowedAfterTriangleBracket})`);
+
+export function validateMarkup(content, applyRegex) {
+	if (content) {
+		if (content.map) return content.map(item => validateMarkup(item));
+		if (content._markup) return content;
+		if (Object.hasOwn(content, '_$litType$')) throw markupError;
+		if (applyRegex && content.constructor === String && disallowedTagsRegex.test(content)) throw markupError;
+	}
+	return content;
+}
+
+export function markup(strings, ...expressions) {
+	strings.forEach(str => validateMarkup(str, true));
+	expressions.forEach(exp => validateMarkup(exp, true));
+	return { ...html(strings, ...expressions), _markup: true };
+}
+
+export const linkGenerator = ({ href, target }) => {
+	import('../components/link/link.js');
+	return chunks => markup`<d2l-link href="${ifDefined(href)}" target="${ifDefined(target)}">${chunks}</d2l-link>`;
+};
```

Comment:
We likely want to add a helper for tooltip-help, something like:

```js
export const tooltipHelpGenerator = ({ text }) => {
	import('../components/tooltip/tooltip-help.js');
	return chunks => markup`<d2l-tooltip-help text="${ifDefined(text)}">${chunks}</d2l-tooltip-help>`;
};
```

P.S. I'm surprised that for tooltip-help, the `text` is what's displayed inline, and the slot contents are in the tooltip - would have guessed the other way around!